src/Microsoft.Health.Dicom.Core/Models/Indexing/ReindexCheckpoint.cs

```diff
@@ -0,0 +1,38 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Microsoft.Health.Dicom.Core.Features.Model;
+using Microsoft.Health.Operations;
+
+namespace Microsoft.Health.Dicom.Core.Models.Indexing;
+
+internal class ReindexCheckpoint : ReindexInput, IOperationCheckpoint
```

Comment:
we should figure out a way to test such change to avoid breaking existing operations.